src/Repository/WorklogRepository.php

```diff
@@ -113,7 +113,7 @@ public function findWorklogsByWorkerAndDateRange(string $workerIdentifier, \Date
             ->getQuery()->getResult();
     }
 
-    public function findBillableWorklogsByWorkerAndDateRange(string $workerIdentifier, \DateTime $dateFrom, \DateTime $dateTo)
+    public function findBillableWorklogsByWorkerAndDateRange(\DateTime $dateFrom, \DateTime $dateTo, ?string $workerIdentifier = null)
```

Comment:
I was thinking something along these lines:

```
public function findBillableWorklogsByWorkerAndDateRange(\DateTime $dateFrom, \DateTime $dateTo, ?string $workerIdentifier = null, bool? $isBilled = null)
{
  ...
 
  if ($isBilled !== null) {
    $qb->andWhere($qb->expr()->eq('worklog.isBilled', ':isBilled'))
    ->setParameter('isBilled', $isBilled);
  }

}        
```


And then in BillableUnbilledHoursReportService call 

```
$billableWorklogs = $this->worklogRepository->findBillableWorklogsByWorkerAndDateRange($dateFrom, $dateTo, null, true);
```

Then the 
```
   if (false === $billableWorklog->isBilled()) {
```
would not be needed in BillableUnbilledHoursReportService. And you would process fewer entities.